sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -141,8 +140,7 @@ case class UnresolvedTableValuedFunction(
  */
 case class UnresolvedAttribute(nameParts: Seq[String]) extends Attribute with Unevaluable {
 
-  def name: String =
-    nameParts.map(n => if (n.contains(".")) s"`$n`" else n).mkString(".")
+  def name: String = nameParts.map(n => if (n.contains(".")) s"`$n`" else n).mkString(".")
```

Comment:
I have a question here: 
 For```UnresolvedAttribute("`a.b`"::"c.d"::Nil).sql```, shall we return
 ```
`a.b`.`c.d`
``` 
instead of
```
```a.b```.`c.d`
```

Reply:
With this change, ```UnresolvedAttribute("`a.b`"::"c.d"::Nil).sql``` returns
```
``a.b``.`c.d`
```
Not
```
```a.b```.`c.d`
```
And I think this is correct because ``` ` ``` should be quoted.

I mistakenly change this line, which is not `sql` but is `name`. As you see, there is no effective change here.
I'll revert the change of this line.
I think `name` also has similar issue bat I'd like focus `sql` in this PR.

Reply:
I think, this is correct.
If a backtick is a part of attribute name, it should be escaped with an additional backtick.
Then, an attribute name which contains escaped backticks should be enclosed by backticks.
MySQL handles such case with the same manner.
https://dev.mysql.com/doc/refman/8.0/en/identifiers.html

Reply:
One example.
If we'd like to express an attribute ``` `a.b` ``` where two backticks are parts of the attribute name.
```
spark.sql("select 'Hello' as `a.b`").show
```
is incorrect because the attribute name will be `a.b`.

```
spark.sql("select 'Hello' as `he``llo`").show
```
is correct and this quote style is what `UnresolvedAttribute.sql` provides with this change.

Reply:
Thanks for the explanation. I see your points.
I have one more question :) 
Should ```($"`a.b`.c".expr.sql)```behave the same as ```UnresolvedAttribute("`a.b`"::"c"::Nil)```?
```($"`a.b`.c".expr.sql)``` returns 
```
`a.b`.`c`
``` 
and ```UnresolvedAttribute("`a.b`"::"c"::Nil)``` returns
```
```a.b```.`c`
```

Reply:
I think it's correct that `($"`a.b`.c".expr.sql)` returns ``` `a.b`.`c` ```.
For `UnresolvedAttribute`, we can specify each name part split by `::` so it's not necessary to quote them.
For `$"name"` or `col()`, we can just pass single string so we need to quote manually.

By the way, I've already noticed that we can't write like
```
$```a.b``` or col(```a.b```)
```
because `UnresolvedAttribute.parseAttributeName` doesn't handle escaped names properly.
I believe it's a bug but it's out of scope of this PR and I'll fix it in another PR.

Reply:
Agree. Thanks for the explanation.